lib/dor/datastreams/content_metadata_ds.rb

```diff
@@ -35,10 +35,40 @@ class ContentMetadataDS < ActiveFedora::OmDatastream
 
     def public_xml
       result = ng_xml.clone
-      result.xpath('/contentMetadata/resource[not(file[(@deliver="yes" or @publish="yes")])]'   ).each { |n| n.remove }
-      result.xpath('/contentMetadata/resource/file[not(@deliver="yes" or @publish="yes")]'      ).each { |n| n.remove }
-      result.xpath('/contentMetadata/resource/file').xpath('@preserve|@shelve|@publish|@deliver').each { |n| n.remove }
-      result.xpath('/contentMetadata/resource/file/checksum').each { |n| n.remove }
+      
+      # remove any resources or attributes that are not destined for the public XML
+      result.xpath('/contentMetadata/resource[not(file[(@deliver="yes" or @publish="yes")]|externalFile)]').each { |n| n.remove }
+      result.xpath('/contentMetadata/resource/file[not(@deliver="yes" or @publish="yes")]'                ).each { |n| n.remove }
+      result.xpath('/contentMetadata/resource/file').xpath('@preserve|@shelve|@publish|@deliver'          ).each { |n| n.remove }
+      result.xpath('/contentMetadata/resource/file/checksum'                                              ).each { |n| n.remove }
```

Comment:
I'm asking about code coverage because coveralls is not enabled on this repo and I don't understand the implications of these XML deletions and whether or not the specs cover all the scenarios these deletions are designed to handle.